nav2_docking/opennav_docking/src/controller.cpp

```diff
@@ -117,6 +125,23 @@ bool Controller::computeVelocityCommand(
   return isTrajectoryCollisionFree(pose, is_docking, backward);
 }
 
+geometry_msgs::msg::Twist Controller::computeRotateToHeadingCommand(
+  const double & angular_distance_to_heading,
+  const geometry_msgs::msg::Twist & current_velocity,
+  const double & dt)
+{
+  geometry_msgs::msg::Twist cmd_vel;
```

Comment:
Can we use the angular acceleration and target orientation to accelerate / decelerate towards the goal? https://github.com/ros-navigation/navigation2/blob/main/nav2_behaviors/plugins/spin.cpp#L140

That way we slow down towards the goal instead of a sudden jerk when we stop and reach the appropriate angle. That'll introduce less odometry slippage